### PR TITLE
Add records

### PR DESCRIPTION
**Pull request by snovikov**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -0,0 +1,16 @@
+locals {
+  records = { for r in var.records : "${r.type}_${r.name}.${var.domain}" => r }
+}
+
+resource "cloudflare_record" "this" {
+  for_each = local.records
+
+  zone_id = data.cloudflare_zone.this.id
+  name    = "${each.value.name}.${var.domain}."
```
Comment (cytopia) [inline main.tf:9]: Are you sure this is not simply:
```
name = each.value.name
```
The domain itself is already identified by the `zone_id`

Comment (snovikov) [reply main.tf:9]: I think this can be defined in two ways. both are correct

Comment (snovikov) [reply main.tf]: changed to `each.value.name`

**Review by titanlien — APPROVED** [review]

File: modules/record_collection/main.tf [hunk 2]
```diff
@@ -0,0 +1,13 @@
+resource "cloudflare_record" "this" {
+  count = length(var.values)
```
Comment (danvaida) [inline modules/record_collection/main.tf:2]: I wonder if this expression won't cause issues when parsing a list of ordered items, should the list change. I recall that this used to be a problem in the past. Surely using a `for_each` on a set was considered here, no?

**Pull request merged**